### PR TITLE
fix: add additional permissions required

### LINKED ISSUE #227: Fix permissions

### PR DESCRIPTION
**Pull request by gregoryduckworth**
Closes #227

### CODE REVIEW

File: .github/workflows/check-linked-issues.yml [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-  pull_request_target:
+  pull_request:
     types: [opened, edited, reopened, synchronize]
 
 jobs:
   check_pull_requests:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      issues: write
```
Comment (simoneb) [inline .github/workflows/check-linked-issues.yml:12]: are you sure that issues need write access? this action is posting comments on pull requests only, as far as I remember

File: .github/workflows/check-linked-issues.yml [hunk 2]
```diff
@@ -1,12 +1,16 @@
 name: Check linked issues
 
 on:
-  pull_request_target:
+  pull_request:
```
Comment (simoneb) [inline .github/workflows/check-linked-issues.yml:4]: why this change? iirc the reason why we recommend using this this way is because it limits what the workflow can do. pull request target works in the context of the base, rather than the source code contained in the PR branch, meaning that it limits potential malicious actions

**Review by simoneb — COMMENTED** [review]
@gregoryduckworth see inline comments. also, we need to decide if this is the way we want to go, because it's considerable changes. if we do go with this, we need to document this in all the actions' repositories

File: .github/workflows/check-linked-issues.yml (hunk 1 above)
Comment (gregoryduckworth) [reply .github/workflows/check-linked-issues.yml:12]: I've looked through the source code and come across https://github.com/nearform/github-action-check-linked-issues/blob/496a217059004f808e81227029686db27c825378/src/action.js#L83 which needs permission to delete a comment/issue, hence the `write` permission.
